### ISSUE TITLE
Uncaught (in promise) DOMException: Failed to execute 'setAttribute' on 'Element'

### ISSUE THREAD

**Issue by jazzmind** [tinymce/tinymce#9552]
## 📝 Provide detailed reproduction steps (if any)
Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud) or similar.
1. initialize tinyMCE on an element containing the following invalid HTML: `<i gothic",="" sans-serif;"="">test</i>`

### ✔️ Expected result
TinyMCE should load and ignore the invalid attributes, potentially stripping them out (I looked for a content filter option to strip invalid attributes but couldn't find one)

### ❌ Actual result
tinymce.min.js?1712860561:4 Uncaught (in promise) DOMException: Failed to execute 'setAttribute' on 'Element': 'gothic",' is not a valid attribute name.

  
### ❓ Possible solution

The issue seems to stem from lines 1173-1180: 

 const rawSet = (dom, key, value) => {
    if (isString(value) || isBoolean(value) || isNumber(value)) {
      **try { dom.setAttribute(key, value + ''); } catch (e) {}**
    } else {
      console.error('Invalid call to Attribute.set. Key ', key, ':: Value ', value, ':: Element ', dom);
      throw new Error('Attribute value was not simple');
    }
  };

by wrapping dom.setAttribute in a try/catch it simply strips the invalid attributes from code and then proceeds to load.

**Comment by TheSpyder** [comment]
Interesting. We use the browser to parse, and these are actually the attributes it gives us.
```new DOMParser().parseFromString('<i gothic",="" sans-serif;"="">test</i>', 'text/html')```

And then `getAttributeNames()` on the `i` tag returns `['gothic",', 'sans-serif;"']`

I'm surprised the browser doesn't strip them, as it is returning names that fail the `setAttribute` call. A try/catch there shouldn't strip those attributes, but stripping them does seem like what we should be doing.

**Comment by jazzmind** [comment]
I think there are two issues here:
1) Without the above try catch wrapper the error thrown by setAttribute breaks the script execution in chrome. This prevents tinymce.init() from completing so I just get the three loading dots and no error messages. I've tried using .catch() and wrapping in a try/catch block but I can't prevent execution from breaking.

2) The browser's own parser seems inconsistent when it comes to invalid attributes. It parses it without error via DOMParser but throws an unrecoverable error via setAttribute. 

Given that inconsistency tinymce probably needs to strip invalid attributes or find another way to fail gracefully without breaking execution. 

Please let me know if i'm missing something obvious in regard to how to catch the error at init time without the execution flow breaking as a temp solution.

**Comment by TheSpyder** [comment]
No, you're right, it's something we have to fix in TinyMCE. I can't see any way for you to work around the runtime error without applying regex to the raw source string.

**Comment by yyt-do** [comment]
how to deal

**Comment by yyt-do** [comment]
This problem does not appear in the version of the editor，how to deal @TheSpyder

**Comment by spocke** [comment]
That is pretty strange the spec clearly states that the `"` should not be a valid character for attribute names:

"Attributes have a name and a value. Attribute names must consist of one or more characters other than [controls](https://infra.spec.whatwg.org/#control), U+0020 SPACE, U+0022 ("), U+0027 ('), U+003E (>), U+002F (/), U+003D (=), and [noncharacters](https://infra.spec.whatwg.org/#noncharacter). In the HTML syntax, attribute names, even those for [foreign elements](https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements), may be written with any mix of [ASCII lower](https://infra.spec.whatwg.org/#ascii-lower-alpha) and [ASCII upper alphas](https://infra.spec.whatwg.org/#ascii-upper-alpha)."
Reference:
https://html.spec.whatwg.org/multipage/syntax.html#attributes-2

Then looking though the WebKit source code I can see that this exception is thrown when the attribute name is not a `qualified name`and it points to these specs:
https://www.w3.org/TR/xml/#NT-NameStartChar
https://www.w3.org/TR/xml/#NT-NameChar

So the `Element.setAttribute` is more narrow than the parsing spec. So it's a combination of those two restrictions we need to apply. So `foo-bar` is a valid attribute name but `-foo` is not for example.

**Comment by yyt-do** [comment]
However, the content I received contains these properties. I used the 5.x editor to echo the content and it worked, but the new version of the editor returned an error

**Comment by yyt-do** [comment]
However, the content I received contains these properties. I used the 5.x editor to echo the content and it worked, but the new version of the editor returned an error![Image](https://github.com/user-attachments/assets/f134612e-75b6-44c1-8725-9b1313938541)



> That is pretty strange the spec clearly states that the `"` should not be a valid character for attribute names:
> 
> "Attributes have a name and a value. Attribute names must consist of one or more characters other than [controls](https://infra.spec.whatwg.org/#control), U+0020 SPACE, U+0022 ("), U+0027 ('), U+003E (>), U+002F (/), U+003D (=), and [noncharacters](https://infra.spec.whatwg.org/#noncharacter). In the HTML syntax, attribute names, even those for [foreign elements](https://html.spec.whatwg.org/multipage/syntax.html#foreign-elements), may be written with any mix of [ASCII lower](https://infra.spec.whatwg.org/#ascii-lower-alpha) and [ASCII upper alphas](https://infra.spec.whatwg.org/#ascii-upper-alpha)." Reference: https://html.spec.whatwg.org/multipage/syntax.html#attributes-2
> 
> Then looking though the WebKit source code I can see that this exception is thrown when the attribute name is not a `qualified name`and it points to these specs: https://www.w3.org/TR/xml/#NT-NameStartChar https://www.w3.org/TR/xml/#NT-NameChar
> 
> So the `Element.setAttribute` is more narrow than the parsing spec. So it's a combination of those two restrictions we need to apply. So `foo-bar` is a valid attribute name but `-foo` is not for example.

**Comment by TheSpyder** [comment]
A lot changed after TinyMCE 5. It’s not something we can reverse, we just need to add filtering for the names.

**Comment by yyt-do** [comment]
Hello, which attribute should I use to filter this error for the newer version? I used valid_elements: '*[*]', which echoes the content, but when I use enter to break lines, the &nbsp; between my p tags; It is not rendered, resulting in content crammed together when displayed @TheSpyder

**Comment by TheSpyder** [comment]
This error is caused by invalid HTML. TinyMCE 5 just didn't operate on elements in a way that caused exceptions in this case.

We will consider adding filters to TinyMCE 7 to avoid the exception, but this will change the HTML (by removing the invalid attribute). You could work around the exception by changing the HTML yourself.

**Comment by DavidDev21** [comment]
Is there any update on this issue? Any fixes or workaround?

**Comment by TheSpyder** [comment]
We tried to fix it with a sanitisation approach and weren't happy with the solution. An alternative proposal internally seems to have fallen off the schedule, I'll poke the appropriate people.

**Comment by MitchC1999** [comment]
We are closing this ticket as the described issue is no longer replicable https://fiddle.tiny.cloud/ipCOmrn2hV/0.
